src/app/article/page.tsx

```diff
@@ -0,0 +1,32 @@
+'use client';
+
+import React from 'react';
+import { useNewsContext } from '@/context';
+import { NewsWithId } from '@/models/types';
+import getHumanDate from '@/utils/getHumanDate';
+import styles from './ArticlePage.module.scss';
+
+type Props = {
+  searchParams: { id: string };
+};
+
+function ArticlePage({ searchParams }: Props) {
+  const { context } = useNewsContext();
+  const article = context?.find((el) => el.id === searchParams.id) as NewsWithId;
```

Comment:
Please remove `as NewsWithId`. In this case when a user refreshes a page there is no context, so the user gets an error

Reply:
Find can return NewsWithId or undefined. When user refreshes the page there is no context, and article is undefined, that's why error appears.